docs/source/tutorials/sveltekit.mdx

```diff
@@ -0,0 +1,431 @@
+   After they have uploaded, scroll down to the button and click "Commit changes to main".
+
+**That's it!** Your application should now be live at `https://huggingface.co/spaces/<your-username>/<your-space-name>`!
+
+## Server-side inference
+
+For server-side inference, we'll rely on [SvelteKit API routes](https://kit.svelte.dev/docs/routing#server) to handle requests.
+
+### Step 1: Initialise the project
+
+Start by creating a new Seveltkit .js application using `npm create svelte@latest`:
+
+```bash
+npm create svelte@latest sveltekit-static
+```
+
+For this demo, we'll be selecting a simple project skeleton without TypeScript and additional packages
+for ESLint, Prettier and later adding Tailwind CSS following the [official guide](https://tailwindcss.com/docs/guides/sveltekit).
+
+### Step 2: Install SvelteKit Node Server Adapter
+
+In this step we'll setup [@sveltejs/adapter-node](https://kit.svelte.dev/docs/adapter-node)
+to change the deployment target to Node.js, so we can build and deploy our application as a Node.js server.
+
+```bash
+npm i -D @sveltejs/adapter-node
+```
+
+Update the `svelte.config.js` file to use the [`adapter-node`](https://kit.svelte.dev/docs/adapter-node):
+
+```js
+import adapter from "@sveltejs/adapter-node";
+import { vitePreprocess } from "@sveltejs/kit/vite";
+/** @type {import('@sveltejs/kit').Config} */
+const config = {
+  kit: {
+    adapter: adapter({
+      // default options are shown
+      out: "build",
+      precompress: false,
+      envPrefix: "",
+      polyfill: true,
+    }),
+  },
+  preprocess: vitePreprocess(),
+};
+export default config;
+```
+
+### Step 3: Install and configure Transformers.js
+
+You can install Transformers.js from [NPM](https://www.npmjs.com/package/@xenova/transformers) with the following command:
+
+```bash
+npm i @xenova/transformers
+```
+
+`./src/lib/server/pipeline.js` - to handle the construction of our pipeline.
+
+```js
+import { pipeline } from "@xenova/transformers";
+
+// Use the Singleton pattern to enable lazy construction of the pipeline.
+// NOTE: We wrap the class in a function to prevent code duplication (see below).
+const P = () =>
+  class PipelineSingleton {
+    static task = "text-classification";
+    static model = "Xenova/distilbert-base-uncased-finetuned-sst-2-english";
+    static instance = null;
+
+    static async getInstance(progress_callback = null) {
+      if (this.instance === null) {
+        this.instance = pipeline(this.task, this.model, {
+          progress_callback,
+        });
+      }
+      return this.instance;
+    }
+  };
+
+let PipelineSingleton;
+if (process.env.NODE_ENV !== "production") {
+  // When running in development mode, attach the pipeline to the
+  // global object so that it's preserved between hot reloads.
+  // For more information, see https://vercel.com/guides/nextjs-prisma-postgres
```

Comment:
Is this section also necessary for Sveltekit? It was originally needed for Next.js (which uses webpack) due to how they handle hot reloads.

Reply:
great question, I'm not sure how vite.js is handling the hotreload here, it might not be needed! I'll remove and see thanks